tests/unittests/EntryPoint.cpp

```diff
@@ -14,22 +14,39 @@
 //
 //******************************************************************************
 
+#ifdef _WIN32
 #include <windows.h>
+#endif
 
 #include <TestFramework.h>
 
 int main(int argc, char** argv) {
+#ifdef _WIN32
     // Initialize COM for all of the tests
     ::CoInitializeEx(NULL, COINIT_MULTITHREADED);
+#endif
 
     testing::InitGoogleTest(&argc, argv);
 
+#ifdef _WIN32
     LOG_INFO("Starting unit tests...");
+#else
+    printf("Starting unit tests...\n");
+#endif
 
     auto result = RUN_ALL_TESTS();
 
+#ifdef _WIN32
     // TODO: move to a scopeguard when we have one in our codebase
     ::CoUninitialize();
+#endif
 
     return result;
 }
+
+#ifndef _WIN32 // Extra declarations for OSX
+extern "C" void WinObjC_SetMissingSelectorFatal(int x) {
+}
+extern "C" void WinObjC_SetZombiesEnabled(int x) {
+}
+#endif
```

Comment:
since we have so many _WIN32s in here, would it make sense to just have one and switch main() ?